src/server/AccountsServer.js

```diff
@@ -15,17 +27,38 @@ class Accounts extends AccountsCommon {
     }
     this.db = db;
   }
+  async createUser(user: UserCreationInputType): Promise<string> {
+    if (this.validateUsername(user.username, false) && this.validateEmail(user.email, false)) {
+      throw new AccountsError({ message: 'Username or Email is required' });
+    }
+    if (user.username && await this.db.findUserByUsername(user.username)) {
```

Comment:
I agree. I was following Meteor's documentation which has the findUserByUsername/Email methods. However your suggested approach makes more sense.